### PR TITLE
Fix AzureCommentReporter where repo names contain spaces

### LINKED ISSUE #4184: [Beta] AzureCommentReporter fails when target repo contains spaces
**Describe the bug**
Following #4132 being resolved in beta branch I've been doing some additional testing and found it still fails in the PR target repository contains spaces

**Expected behavior**
PR comment to be created on PR target repos that contain spaces

**Screenshots**
image

**Additional context**
The `SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI` environment variable will pass the target repository name with %20 URL encoding instead of spaces so these need to be replaced to get the correct repository name.

### PR DESCRIPTION
**Pull request by lukelloyd1985**
Fixes #4184 

## Proposed Changes

When using `SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI` variable any repositories that have spaces in the name will need the `%20` URL encoded space converted back to a space.

### Author/Contributor
- [x] Add entry to the [CHANGELOG](https://github.com/oxsecurity/megalinter/blob/main/CHANGELOG.md) listing the change and linking to the corresponding issue (if appropriate)
- [x] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

File: megalinter/reporters/AzureCommentReporter.py [hunk 1]
```diff
@@ -116,7 +116,7 @@ def produce_report(self):
                     + "See https://learn.microsoft.com/en-us/azure/devops/pipelines/"
                     + "build/variables?view=azure-devops&tabs=yaml"
                 )
-                repository_name = SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI.split("/")[-1]
+                repository_name = SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI.split("/")[-1].replace('%20', ' ')
```
Comment (echoix) [inline megalinter/reporters/AzureCommentReporter.py:119]: Is it only spaces that need to be encoded? Or a general urlencoding should be used?

Comment (lukelloyd1985) [reply megalinter/reporters/AzureCommentReporter.py:119]: It's more decoding from the URL and spaces should be sufficient based on the ADO imposed restrictions?

https://learn.microsoft.com/en-us/azure/devops/organizations/settings/naming-restrictions?view=azure-devops#azure-repos-git

**Comment by lukelloyd1985** [comment]
Can this PR be merged please?

**Comment by nvuillam** [comment]
@lukelloyd1985 please can you test with beta and confirm it's ok ? :)

**Pull request merged**